src/chunktransformer.js

```diff
@@ -0,0 +1,390 @@
+      })
+    );
+    return;
+  }
+  if (typeof chunk.rowKey === 'undefined' || chunk.rowKey === '') {
+    this.destroy(
+      new TransformError({
+        message: 'A row key must be set',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  if (chunk.resetRow) {
+    this.destroy(
+      new TransformError({
+        message: 'A new row cannot be reset',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  if (prevRowKey === newRowKey) {
+    this.destroy(
+      new TransformError({
+        message: 'A commit happened but the same key followed',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  if (!chunk.familyName) {
+    this.destroy(
+      new TransformError({
+        message: 'A family must be set',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  if (!chunk.qualifier) {
+    this.destroy(
+      new TransformError({
+        message: 'A column qualifier must be set',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  this.validateValueSizeAndCommitRow(chunk);
+};
+/**
+ * Validates state for rowInProgress
+ * @private
+ * @param {chunk} chunk chunk to validate
+ */
+ChunkTransformer.prototype.validateRowInProgress = function(chunk) {
+  const row = this.row;
+  const newRowKey = Mutation.convertFromBytes(chunk.rowKey);
+  if (chunk.rowKey && newRowKey !== row.key) {
+    this.destroy(
+      new TransformError({
+        message: 'A commit is required between row keys',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  if (chunk.familyName && !chunk.qualifier) {
+    this.destroy(
+      new TransformError({
+        message: 'A qualifier must be specified',
+        chunk: chunk,
+      })
+    );
+    return;
+  }
+  this.validateResetRow(chunk);
+  this.validateValueSizeAndCommitRow(chunk);
+};
+/**
+ * Validates chunk for cellInProgress state.
+ * @private
+ * @param {chunk} chunk chunk to validate
+ */
+ChunkTransformer.prototype.validateCellInProgress = function(chunk) {
+  this.validateResetRow(chunk);
+  this.validateValueSizeAndCommitRow(chunk);
+};
+/**
+ * Moves to next state in processing.
+ * @private
+ * @param {chunk} chunk chunk in process
+ */
+ChunkTransformer.prototype.moveToNextState = function(chunk) {
+  const row = this.row;
+  if (chunk.commitRow) {
+    this.push(row);
+    this.commit();
+  } else {
+    if (chunk.valueSize > 0) {
+      this.state = RowStateEnum.CELL_IN_PROGRESS;
+    } else {
+      this.state = RowStateEnum.ROW_IN_PROGRESS;
+    }
+  }
+};
+/**
+ * Process chunk when in NEW_ROW state.
+ * @private
+ * @param {chunks} chunk chunk to process
+ * @param {options} options options
+ * @param {callback} callback callback to call with row as and when generates
+ * @returns {boolean} return false to stop further processing.
+ */
+ChunkTransformer.prototype.processNewRow = function(chunk) {
+  const newRowKey = Mutation.convertFromBytes(chunk.rowKey);
+  this.validateNewRow(chunk);
+  if (this._destroyed) {
```

Comment:
I think we can get rid of this. We do the same check in the transform function.

Reply:
validateNewRow has a check for existence of familyName and chunk.qualifier in chunk. which I later use it without checking.  

without check
```javascript
this.family = row.data[chunk.familyName.value] = {};
  const qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
```

Do you suggest to check for existence of that instead?
```javascript
  if (chunk.familyName && chunk.qualifier) {
    this.family = row.data[chunk.familyName.value] = {};
    const qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
    this.qualifiers = this.family[qualifierName] = [];
    this.qualifier = {
      value: Mutation.convertFromBytes(chunk.value, this.options),
      labels: chunk.labels,
      timestamp: chunk.timestampMicros,
    };
    this.qualifiers.push(this.qualifier);
    this.moveToNextState(chunk);
  }
```

Reply:
I just meant we should be able to get rid of the check if the stream was destroyed, as we do that same check from the transform function.

Reply:
I updated PR as per your latest feedback except this one to get rid of checks completely. I replaced the check this._destroyed with chunk.familyName && chunk.qualifier otherwise its throwing error bcz of this statement
```javascript
this.family = row.data[chunk.familyName.value] = {};
    const qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
```